apps/lenra_web/lib/lenra_web/router.ex

```diff
@@ -13,82 +13,121 @@ defmodule LenraWeb.Router do
+    plug(Plug.ExtractBearer)
+    plug(Plug.VerifyScope, "manage:apps")
+  end
+
+  pipeline :scope_store do
+    plug(Plug.ExtractBearer)
+    plug(Plug.VerifyScope, "store")
+  end
+
+  pipeline :scope_resources do
+    plug(Plug.ExtractQueryParams)
+    plug(Plug.VerifyScope, "resources")
+  end
+
+  # remains of Authentication API. Move it to /api ?
   scope "/auth", LenraWeb do
-    pipe_through(:api)
-    post("/register", UserController, :register)
-    post("/login", UserController, :login)
+    pipe_through([:api])
     post("/password/lost", UserController, :send_lost_password_code)
     put("/password/lost", UserController, :change_lost_password)
 
-    pipe_through(:ensure_auth_refresh)
-    post("/logout", UserController, :logout)
-
-    pipe_through([:ensure_cgu_accepted])
-    post("/refresh", UserController, :refresh_token)
+    pipe_through([:scope_none])
+    get("/me", UserController, :current_user)
   end
 
+  # Runner callback, secured via runner-specific token
   scope "/runner", LenraWeb do
     pipe_through([:api, :runner])
     put("/builds/:id", RunnerController, :update_build)
   end
 
+  # /api, No scope needed
   scope "/api", LenraWeb do
     pipe_through([:api])
     get("/cgu/latest", CguController, :get_latest_cgu)
     get("/apps/:app_service_name", AppsController, :get_app_by_service_name)
+  end
 
-    pipe_through([:ensure_auth])
+  # /api, scope "manage_account"
+  scope "/api", LenraWeb do
+    # Without CGU Accepted
+    pipe_through([:api, :scope_manage_account])
     post("/cgu/:cgu_id/accept", CguController, :accept)
     get("/cgu/me/accepted_latest", CguController, :user_accepted_latest_cgu)
 
+    # Adding CGU for the rest
     pipe_through([:ensure_cgu_accepted])
     post("/verify", UserController, :validate_user)
     post("/verify/lost", UserController, :resend_registration_token)
-    resources("/apps", AppsController, only: [:index, :create, :update, :delete])
-    get("/apps/:app_id/main_environment", ApplicationMainEnvController, :index)
-    resources("/apps/:app_id/environments", EnvsController, only: [:index, :create])
-    patch("/apps/:app_id/environments/:env_id", EnvsController, :update)
-
-    resources("/apps/:app_id/environments/:env_id/invitations", UserEnvironmentAccessController, only: [:index, :create])
-
-    get("/apps/invitations/:id", UserEnvironmentAccessController, :fetch_one)
-    post("/apps/invitations/:id", UserEnvironmentAccessController, :accept)
 
-    resources("/apps/:app_id/builds", BuildsController, only: [:index, :create])
-
-    resources("/apps/deployments", DeploymentsController, only: [:create])
-    get("/apps/:app_id/deployments", DeploymentsController, :index)
     put("/password", UserController, :change_password)
     put("/verify/dev", UserController, :validate_dev)
 
+    get("/webhooks", WebhooksController, :index)
+    post("/webhooks", WebhooksController, :api_create)
+  end
+
+  # /api, scope "store" & CGU Accepted
+  scope "/api", LenraWeb do
```

Comment:
I would say, use the `store` scope in the backoffice client

Reply:
Does it have the same behavior ?
I don't understand why we would list the apps the current user can run in the backoffice ?
I think two different endpoints might be the best solution.

Reply:
> Does it have the same behavior ?

Yes but re-using the scope is WAY cleaner than duplicate the scope or route.

> I don't understand why we would list the apps the current user can run in the backoffice ?

I don't know, but if it's needed, there is no reason not to use it.

> I think two different endpoints might be the best solution.

I strongly disagree, the endpoints should NEVER depend to the client. We NEVER should duplicate a same endpoint and consume it from two different apps. The API shoud be designed independently to the needs of the clients.
Same thing, the scopes should designed independently. It's absolutely NOT an issue to ask for a specific scope if the feature is needed. In contrary, it IS an issue to duplicate a scope just to adapt to a client need.

The only acceptable change IMO is to re-design the scope to create more granularity to avoid giving too much permission to the backoffice if it's an issue. But i don't think it's an issue at all since it was already the case (and worse) before the scope system.

Reply:
I agree to what you say, but I think that this endpoint should not be used in the backoffice and we probably use it for a bad reason.

Reply:
Maybe, it should be investigated. For now, i would say use the "store" scope and open an issue to fix that later ! :)